server/lib/chat/chat-service.ts

```diff
@@ -589,20 +596,31 @@ export default class ChatService {
             userId: msg.userId,
             time: Number(msg.sent),
           })
-          users.push({ id: msg.userId, name: msg.userName })
+          userIds.add(msg.userId)
           break
 
         default:
           return assertUnreachable(msg.data)
       }
     }
 
-    const mentions = await findUsersById(Array.from(mentionIds))
+    const [users, userMentions, channelMentions] = await Promise.all([
+      findUsersById(Array.from(userIds)),
+      findUsersById(Array.from(userMentionIds)),
+      this.getChannelInfos(Array.from(channelMentionIds), userId),
+    ])
+
+    const deletedChannels = subtract(
```

Comment:
Given the general case is "no channels are deleted", I would probably write this as:
```ts
const deletedChannels = channelMentionIds.length === channelMentions.size ? [] : Array.from(subtract(channelMentionIds, channelMentions.map(c => c.id)))
```

No sense in iterating the whole set/map if we already know it's all there